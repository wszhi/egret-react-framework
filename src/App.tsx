import React from 'react'
import { HashRouter as Router, Link, Route, Switch } from 'react-router-dom'
import { GamePage, HomePage } from './pages'
import Game from '@/game'

class App extends React.Component {
  render() {
    return (
      <Router>
        <ul>
          <li>
            <Link to="/home">Home Page</Link>
          </li>
          <li>
            <Link to="/game">Game Page</Link>
          </li>
        </ul>
        <Switch>
          <Route path="/game" key="/game" component={GamePage} />
          <Route path="/home" key="/home" component={HomePage} />
        </Switch>
        <Game />
      </Router>
    )
  }
}

export default App
