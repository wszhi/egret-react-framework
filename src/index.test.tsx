import { shallow } from 'enzyme'
import React from 'react'
import App from './App'

describe('App component test', () => {
  let wrapper: any

  it('snapshot test', () => {
    wrapper = shallow(<App />)
    expect(wrapper.render()).toMatchSnapshot()
  })
})
