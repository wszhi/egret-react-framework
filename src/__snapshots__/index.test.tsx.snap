// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test snapshot test 1`] = `
<ul>
  <li>
    <a
      href="#/home"
    >
      Home Page
    </a>
  </li>
  <li>
    <a
      href="#/game"
    >
      Game Page
    </a>
  </li>
</ul>
`;
